internal/email/email_test.go

```diff
@@ -0,0 +1,167 @@
+// SPDX-FileCopyrightText: Copyright 2024 The Minder Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package email
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func TestIsValidField(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		input          string
+		expectedErr    bool
+		expectedErrMsg string
+	}{
+		// Test case 1: Valid plain text
+		{"Just plain text", false, ""},
+
+		// Test case 2: String with HTML tags
+		{"<b>Bold Text</b>", true, "string <b>Bold Text</b> contains HTML injection"},
+
+		// Test case 3: String with HTML entity
+		{"This is a test &amp; example.", true, "string This is a test &amp; example. contains HTML injection"},
+
+		// Test case 4: String with multiple HTML entities
+		{"This &amp; that &lt; should &gt; work.", true, "string This &amp; that &lt; should &gt; work. contains HTML injection"},
+
+		// Test case 5: Valid URL (no HTML or JavaScript injection)
+		{"https://example.com", false, ""},
+
+		// Test case 6: Mixed content with HTML and JS
+		{"Hello <b>World</b> onload=alert('test');", true, "string Hello <b>World</b> onload=alert('test'); contains HTML injection"},
+
+		// Test case 7: HTML-style comment
+		{"<!-- This is a comment -->", true, "string <!-- This is a comment --> contains HTML injection"},
+
+		// Test case 8: ensure allowed length is less than 200 characters
+		{strings.Repeat("a", MaxFieldLength+1), true, fmt.Sprintf("field value %s is more than %d characters", strings.Repeat("a", MaxFieldLength+1), MaxFieldLength)},
+	}
+
+	for _, tt := range tests {
+		tt := tt // capture range variable for parallel execution
+		t.Run(tt.input, func(t *testing.T) {
+			t.Parallel()
+			err := isValidField(tt.input)
+			if (err != nil) != tt.expectedErr {
+				t.Errorf("isValidField(%q) got error: %v, expected error: %v", tt.input, err, tt.expectedErr)
+			}
+			if err != nil && err.Error() != tt.expectedErrMsg {
+				t.Errorf("isValidField(%q) got error message: %v, expected message: %v", tt.input, err.Error(), tt.expectedErrMsg)
+			}
+		})
+	}
+}
+
+func TestValidateDataSourceTemplate(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		input          bodyData
+		expectedErr    bool
```

Comment:
Do you need `expectedErr` separately from `expectedErrMsg != ""`?

Reply:
Thanks 👍 Removed the obsolete bool for expected error 👍